internal/platform/platform.go

```diff
@@ -9,6 +9,26 @@ import (
 	"runtime"
 )
 
+// IsSupported is exported for tests and includes constraints here and also the assembler.
+func IsSupported() bool {
+	if _, ok := map[string]struct{}{
```

Comment:
I would prefer switch rather than map as this might end up having allocating map each time or not
```go
switch runtime.GOOS {
case "darwin", "windows", "linux":
default: return false
}
```

Reply:
this way everything can be compile time const so friendlier for Go compiler 😉

Reply:
sure it was test but you are right this is cleaner anyway!